### PR TITLE
Avoid querying bulk_customers for each repo being serialized

### PR DESCRIPTION
**Pull request by salbertson**

### CODE REVIEW

**Comment by gylaz** [comment]
@salbertson Thanks! We were actually thinking that a bulk customers can now be represented by an `Owner` that is `whitelisted`. Thus we can preload a `repo`'s `owner` and check that flag. What do you think about this approach?

**Comment by salbertson** [comment]
@gylaz hmm, not sure. Is "whitelisting" a thing we do now in Hound?

**Comment by salbertson** [comment]
[In RepoSerializer repo.bulk? causes N+1. Let's fix that.](https://trello.com/c/i4m7XIVB/813-in-reposerializer-repobulk-causes-n1-lets-fix-that)

**Comment by gylaz** [comment]
Creating a bulk customer record is basically what we call whitelisting. We could keep the bulk customer table, but make a `belongs_to` relationship to an `Owner`. Then we could do `repo.owner.bulk_customer.present?` to mean it's bulk/whitelisted.

File: app/controllers/repos_controller.rb [hunk 1]
```diff
@@ -10,8 +10,17 @@ def index
...
+          bulk_customers_by_org: find_bulk_customers(repos).index_by(&:org),
+        )
       end
     end
   end
+
+  private
+
+  def find_bulk_customers(repos)
+    BulkCustomer.where(org: repos.map(&:organization).uniq)
```
Comment (gylaz) [inline app/controllers/repos_controller.rb]: `map` -> `pluck`?

File: app/serializers/repo_serializer.rb [hunk 2]
```diff
@@ -37,4 +37,16 @@ def has_admin_membership?
   def has_subscription?
     object.subscription&.user_id == scope.id
   end
+
+  def bulk?
+    if bulk_customers_by_org
```
Comment (gylaz) [inline app/serializers/repo_serializer.rb]: In what situation will this be nil?

File: spec/controllers/repos_controller_spec.rb [hunk 3]
```diff
@@ -26,5 +26,25 @@
         expect(user.repos.size).to eq(1)
       end
     end
+
+    context "when current user is a bulk member of a repo" do
+      it "returns 0 price in cents for that repo" do
+        repo = create(
+          :repo,
+          in_organization: true,
+          private: true,
```
Comment (gylaz) [inline spec/controllers/repos_controller_spec.rb]: Would our `in_private_org` trait come-in handy here?

File: spec/controllers/repos_controller_spec.rb [hunk 4]
```diff
@@ -26,5 +26,25 @@
...
+          name: "yeehaw/wat",
+        )
+        create(:bulk_customer, org: "yeehaw")
+        user = create(:user)
+        user.repos << repo
```
Comment (gylaz) [inline spec/controllers/repos_controller_spec.rb]: Could combine these two lines as `user = create(:user, repos: [repo])`.

**Comment by gylaz** [comment]
I've opened #1433 to basically do this, but in a simpler "whitelist" approach. Going to close this PR in favor of the other one.

**Pull request closed without merging**